protocol/messenger_mailserver_cycle.go

```diff
@@ -517,6 +520,7 @@ func (m *Messenger) waitForAvailableStoreNode(timeout time.Duration) bool {
 	wg.Add(1)
 
 	go func() {
+		defer gocommon.LogOnPanicAndRethrow()
 		defer func() {
 			wg.Done()
 		}()
```

Comment:
This could actually be the solution to the problem with go defer stacks.

We could run 2 regexes
- one to check if `go func() {` is followed by `defer gocommon.LogOnPanicAndRethrow()`.
- a second regex to check if `go func() {` is followed by:

```go
defer func() {
    gocommon.LogOnPanicAndRethrow()
    ...
```

Reply:
This way we skip the defer stack and rely on standard function order.

Reply:
> This way we skip the defer stack and rely on standard function order. What do you think @qfrank ?

@Samyoul, the issue you're concerned about is when the other defer panics, right? I understand that we might be more interested in the first panic, but I believe that the latest panic would be sufficient - it's already a significant clue, doesn't it? Also, considering the likelihood of this happening, I would keep it simple and make `defer gocommon.LogOnPanicAndRethrow()` always a first-liner. This approach will make our lives easier both in adapting to the new pattern and managing script complexity.